pkg/menus/menus.go

```diff
@@ -0,0 +1,162 @@
+package menus
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/gizak/termui/v3/widgets"
+)
+
+var menuWidth = 50
+
+// Entry contains all the information needed for a boot entry
+type Entry interface {
+	// Label is the string will show in menu
+	Label() string
+	// IsDefault is true means this entry will be hit by default, if there is many default choise, the first on in the list will be choose. the rest will be ignored.
+	IsDefault() bool
+}
+
+// DisplayMenu will present all entries into a menu with numbers, so that user can choose from them.
+func DisplayMenu(menuTitle string, introwords string, location int, entries []Entry) (int, error) {
+	// open a new window
+	if err := ui.Init(); err != nil {
+		log.Printf("failed to initialize termui: %v", err)
+	}
+	// listData contains all choice's labels
+	listData := []string{}
+
+	for i, e := range entries {
+		listData = append(listData, fmt.Sprintf("[%d] %s", i+1, e.Label()))
```

Comment:
You could also start the menu numbering from 0 if that makes life easier :)

Reply:
yep.

Reply:
Reminder to start the menu numbering from 0!